Detectors/FIT/FV0/simulation/src/Digitizer.cxx

```diff
@@ -116,56 +122,65 @@ void Digitizer::process(const std::vector<o2::fv0::Hit>& hits,
+      Float_t const tPhE = t + mRndSignalShape.getNextValue();
+      Int_t const firstBin = roundVc(TMath::Max((Int_t)0, (Int_t)((tPhE - DigitizationParameters::mPmtTransitTime) / mBinSize)));
       Int_t const lastBin = TMath::Min((Int_t)mNBins - 1, (Int_t)((tPhE + 2. * DigitizationParameters::mPmtTransitTime) / mBinSize));
-      for (Int_t iBin = firstBin; iBin <= lastBin; ++iBin) {
-        Float_t const tempT = mBinSize * (0.5 + iBin) - tPhE;
-        mPmtChargeVsTime.at(detId).at(iBin) += mGainVar.at(mIteratorGainVar) * charge * PmtResponse(tempT);
-      } // time-bin loop
-
-      // RandomRing servicing (perhaps a separate class could do better)
-      mIteratorGainVar++;
-      if (mIteratorGainVar >= mGainVar.size()) {
-        mIteratorGainVar = 0;
+      Float_t tempT = mBinSize * (0.5f + firstBin) - tPhE;
+      Float_t* p = analogSignal.data() + firstBin;
+      Vc::float_v binVc(tempT);
+      binVc += mBinSize * Vc::float_v::IndexesFromZero();
+      Vc::float_v binIncVc(mBinSize * Vc::float_v::Size);
+
+      Vc::float_v workVc;
+      Int_t iBin = firstBin;
+      for (Int_t const lastBinRounded = roundVc(lastBin); iBin < lastBinRounded;) {
+        workVc.load(p);
+        workVc += mRndGainVar.getNextValueVc() * charge * PmtResponseVc(binVc);
+        workVc.store(p);
+        Vc::prefetchForOneRead(p);
+        binVc += binIncVc;
+        p += Vc::float_v::Size;
+        iBin += Vc::float_v::Size;
       }
-      mIteratorSignalShape++;
-      if (mIteratorSignalShape >= mSignalShape.size()) {
-        mIteratorSignalShape = 0;
+      tempT += binVc[0] + mBinSize;
+      // loop over the SIMD leftovers
+      for (Float_t const* pEnd = analogSignal.data() + lastBin; p < pEnd; ++p) {
+        *p += mRndGainVar.getNextValue() * charge * PmtResponse(tempT);
+        tempT += mBinSize;
       }
-    }   //photo electron loop
-    mIteratorScintDelay++;
-    if (mIteratorScintDelay >= mScintillatorDelay.size()) {
-      mIteratorScintDelay = 0;
-    }
+    } //photo electron loop
 
     // Charged particles in MCLabel
-    Int_t parentId = hit.GetTrackID();
+    Int_t const parentId = hit.GetTrackID();
     if (parentId != parentIdPrev) {
       mMCLabels.emplace_back(parentId, mEventId, mSrcId, detId);
       parentIdPrev = parentId;
     }
   } //hit loop
 
   // Sum charge of all time bins to get total charge collected for a given channel
-  int first = digitsCh.size(), nStored = 0;
+  Int_t first = digitsCh.size(), nStored = 0;
   for (Int_t ipmt = 0; ipmt < DigitizationParameters::NCHANNELS; ++ipmt) {
-    Float_t totalCharge = 0;
+    Float_t totalCharge = 0.0f;
     for (Int_t iTimeBin = 0; iTimeBin < mNBins; ++iTimeBin) {
-      Float_t timeBinCharge = mPmtChargeVsTime[ipmt][iTimeBin] / DigitizationParameters::CHARGE_PER_ADC;
+      Float_t const timeBinCharge = mPmtChargeVsTime[ipmt][iTimeBin] / DigitizationParameters::CHARGE_PER_ADC;
```

Comment:
instead of dividing each each charge by the same constant and then adding it is better to add all non-normalized charges and the mutliply the sum by the inverse of ``DigitizationParameters::CHARGE_PER_ADC``

Reply:
Done.